### PR TITLE
Benchmarks: micro benchmarks - add python code for DirectXGPUCoreFlops 

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
add python code for DirectX core flops and init DirectX test pipeline.

**Major Revision**
- add python code for DirectX core flops 
- init DirectX test pipeline

**Minor Revision**
- add test for DirectX core flops

### CODE REVIEW

File: .github/workflows/directx-unit-test.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+name: DirectX unit test
```
Comment (cp5555) [inline .github/workflows/directx-unit-test.yml:1]: Why related to this PR?

File: tests/helper/decorator.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 rocm_test = unittest.skipIf(os.environ.get('SB_TEST_ROCM', '0') == '0', 'Skip ROCm tests.')
```
Comment (cp5555) [inline tests/helper/decorator.py:12]: Maybe separated PR?

File: .github/workflows/directx-unit-test.yml (hunk 1 above)
Comment (yukirora) [reply .github/workflows/directx-unit-test.yml:1]: not related to this PR, just I have no confidence on this pipeline and cannot test it locally, so temporarily leverage this PR to test it, will create another PR for it when it works.

Comment (cp5555) [reply .github/workflows/directx-unit-test.yml:1]: Sure, thanks. I have approved this PR. Please make sure these code has been removed.

**Review by cp5555 — APPROVED** [review]

File: superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    """The DirectXGPUCoreFlops benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+        self._bin_name = 'GPUCore.exe'
```
[the remark this answers is not in the document]
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance.py]: from superbench, updated to DirectXGPUCoreFlops.exe

File: superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        self._support_precisions = ['fp16', 'fp32']
+        self._precision_need_to_run = list()
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+        self._parser.add_argument(
+            '--num_loops',
```
[the remark this answers is not in the document]
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance.py:30]: added

File: superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                    'The result format is invalid - round: {}, benchmark: {}, raw output: {}.'.format(
+                        self._curr_run_index, self._name, raw_output
+                    )
+                )
+                return False
+        self._result.add_result(precision + '_flops', max(flops))
+        return True
+
+
+BenchmarkRegistry.register_benchmark('directx-gpucoreflops', DirectXGPUCoreFlops, platform=Platform.DIRECTX)
```
[the remark this answers is not in the document]
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance.py]: updated

**Review by yzygitzh — APPROVED** [review]

**Pull request merged**